### PR TITLE
Updating packages, imports, references to use latest versions and handle deprecations for tree-widgets

### PR DESCRIPTION
**Pull request by Jake-Screen**
- Updated imports of core packages to 4.0.0-dev versions
- Removed imports and references to deprecated components that were removed
- Updated imports and references to deprecated components to their new versions
- commented out references to SpacialContainmentTree pending further discussion
- updated test snapshots to reflect updated api returns
- removed reference to TreeDataChangeListener, which was removed previously
- increased viable node version for running tests through rush to node 19

### CODE REVIEW

File: packages/itwin/tree-widget/src/components/trees/spatial-tree/SpatialTreeComponent.tsx [hunk 1]
```diff
@@ -1,23 +1,21 @@
-/*---------------------------------------------------------------------------------------------
-* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
-* See LICENSE.md in the project root for license terms and full copyright notice.
-*--------------------------------------------------------------------------------------------*/
-import { SpatialContainmentTree, useActiveIModelConnection } from "@itwin/appui-react";
-import React from "react";
-import type { SpatialTreeProps } from "../../../types";
-import { AutoSizer } from "../../utils/AutoSizer";
+// /*---------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (grigasp) [reply packages/itwin/tree-widget/src/components/trees/spatial-tree/SpatialTreeComponent.tsx:1]: > @iTwin/itwinjs-core-presentation
> 
> I believe the Spatial Tree from core was deprecated in 3.6; but the breakdown trees pkg is abandoned and I would rather not have users point to that one.
> 
> Should we copy the implementation of the spatial tree from [core@3.x](mailto:core@3.x) into here?

This is the component that should replace the Spatial Tree from core: https://github.com/iTwin/viewer-components-react/blob/1100f8abce627af728002c513dc707a0d5f366e2/packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx#L64

File: packages/itwin/tree-widget/package.json [hunk 2]
```diff
@@ -44,27 +44,27 @@
...
-    "@itwin/core-quantity": "^3.5.0",
-    "@itwin/core-react": "^3.5.0",
-    "@itwin/eslint-plugin": "^3.5.0",
-    "@itwin/imodel-components-react": "^3.5.0",
-    "@itwin/presentation-common": "^3.5.0",
-    "@itwin/presentation-components": "^3.5.0",
-    "@itwin/presentation-frontend": "^3.5.0",
-    "@itwin/presentation-testing": "^3.5.0",
-    "@itwin/webgl-compatibility": "^3.5.0",
+    "@itwin/appui-abstract": "4.0.0-dev.22",
```
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply packages/itwin/tree-widget/package.json]: updated versions

File: packages/itwin/tree-widget/package.json [hunk 3]
```diff
@@ -95,15 +95,16 @@
     "sinon-chai": "^3.2.0",
     "typemoq": "^2.1.0",
     "typescript": "~4.3.0",
-    "xmlhttprequest": "^1.8.0"
+    "xmlhttprequest": "^1.8.0",
+    "@itwin/breakdown-trees-react": "^0.3.16"
```
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply packages/itwin/tree-widget/package.json]: removed

**Comment by simasjar** [comment]
I am going to take care of this issue (#388) you can close this pr

**Pull request closed without merging**